README.rst

```diff
@@ -32,58 +35,83 @@ On some networks discovery doesn't work reliably, in that case if you can find t
+A few important notes:
+
+- Not all devices are supported for setup.
+- WeMo requires a password of at least 8 characters long.
+- For a WeMo without internet access, see `this guide <https://github.com/pavoni/pywemo/wiki/WeMo-Cloud#disconnecting-from-the-cloud>`_ to stop any blinking lights.
+- If connecting to an open network, the password argument is ignored and you can provide anything, e.g. ``password=None``.
+- If connecting to an AES-encrypted network, OpenSSL is used to encrypt the password by the ``pywemo`` library.
+  It must be installed and available on your ``PATH`` via calling ``openssl`` from a terminal or command prompt.
 
 Firmware Warning
 ----------------
 Starting in May of 2020, Belkin started requiring users to create an account and login to the app (Android app version 1.25).
 In addition to the account, most of the app functionality now requires a connection to the cloud (internet access), even for simple actions such as toggling a switch.
 All of the commands that go through the cloud are encrypted and cannot be easily inspected.
-This raises the possibility that Belkin could, in the future, update Wemo device firmware and make breaking API changes that can not longer be deciphered.
-If this happens, pywemo may no longer function on that device.
-It would be prudent to upgrade firmware cautiously and preferably only after confirming that breaking API changes have not been introduced.
+This raises the possibility that Belkin could, in the future, update WeMo device firmware and make breaking API changes that can no longer be deciphered.
+If this happens, ``pywemo`` may no longer function on that device.
+Thus it would be prudent to upgrade firmware cautiously and preferably only after confirming that breaking API changes have not been introduced.
 
 Developing
 ----------
-Setup and builds are fully automated. You can run build pipeline locally by running.
+Setup and builds are fully automated.
+You can run the build pipeline locally via:
 
 .. code-block::
 
-    # Setup, build, lint and test the code:
-
+    # setup, install, format, lint, test and build:
     ./scripts/build.sh
 
+Note that this will install a git ``pre-commit`` hook.
+For this hook to work correctly, ``poetry`` needs to be globally accessible on your ``PATH`` or the local virtual environment must be activated.
+This virtual environment can be activated with:
+
+.. code-block::
+
+    . .venv/bin/activate
+
 History
 -------
 This started as a stripped down version of `ouimeaux <https://github.com/iancmcc/ouimeaux>`_, but has since taken its own path.
 
 License
 -------
-Some of the code in `pywemo/ouimeaux_device` was originally written by by Ian McCracken (and is copyright). It is released under the BSD license.
+Some of the code in `pywemo/ouimeaux_device` was originally written, copyright, and released under the BSD license by Ian McCracken.
```

Comment:
Mind if we break this out a bit?
- The code in `pywemo/ouimeaux_device` is governed by the BSD LICENSE file in that directory. Some of this code was originally written and copyright by Ian McCracken.

Reply:
Sure, makes sense.  And more updates to this may come when you do the contributing guide and CLA to make it all more clear.